DFC.ServiceTaxonomy.ApiFunction/Function/Execute.cs

```diff
@@ -87,6 +91,26 @@ public Execute(IOptionsMonitor<ServiceTaxonomyApiSettings> serviceTaxonomyApiSet
             }
         }
 
+        private string BuildHostFunctionUrl(HttpRequest req, ILogger log)
+        {
+            var host = req.Headers["X-Forwarded-Host"].ToString();
+
+            if (string.IsNullOrWhiteSpace(host))
+                throw ApiFunctionException.InternalServerError("X-Forwarded-Host header not present.");
+
+            if (string.IsNullOrWhiteSpace(_serviceTaxonomyApiSettings.CurrentValue.Scheme))
+                throw ApiFunctionException.InternalServerError("Scheme missing in Settings");
+
+            if (string.IsNullOrWhiteSpace(_serviceTaxonomyApiSettings.CurrentValue.ApplicationName))
+                throw ApiFunctionException.InternalServerError("ApplicationName missing in Settings");
+
+            var hostUriBuilder = new UriBuilder { Host = host, Scheme = _serviceTaxonomyApiSettings.CurrentValue.Scheme, Path = $"{_serviceTaxonomyApiSettings.CurrentValue.ApplicationName}/{GetFunctionName()}/Execute/" };
```

Comment:
The function name should be specified in the queries themselves, as we want the initial url fragment to be reusable for specifying urls to other endpoints.
For simplicity, we could hard-code the 'Execute'; in the queries as well.